crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,840 @@
+# Copyright (c) Mysten Labs, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# Open Questions
+# --------------
+# - Rules for nullabilit/non-nullability
```

Comment:
I started making the key fields of certain types non-nullable, but I couldn't do this consistently, and it's also not clear to me whether we should use a logical primary key for this or an opaque ID type that works with connections.

Reply:
I would say we'd probably want both to available as non-nullable fields.  I'd be curious about which key fields can't be non-nullable and why, might be an indicator we have the wrong abstraction in the schema

Reply:
"Couldn't" is maybe too strong a term -- I could but I had doubts about doing it.  Most of the cases were where a type's key was compound in some way (e.g. `NormalizedFunction` has `module: NormalizedModule` and `name: String` as key fields), and part of the key is some other rich type (in this case `module`) -- is it okay to make that `module` non-nullable, or does that make the schema promise too much?